src/RTU.cpp

```diff
@@ -0,0 +1,156 @@
+#include <modbus/RTU.hpp>
+#include <cstring>
+#include <cmath>
+#include <modbus/Exceptions.hpp>
+
+using namespace std;
+using namespace base;
+using namespace modbus;
+
+base::Time RTU::interframeDuration(int bitrate) {
+    int duration_us = ceil(1000000.0 / static_cast<double>(bitrate) *
+                           SERIAL_BITS_PER_CHAR * 3.5);
+    return base::Time::fromMicroseconds(max(duration_us, 1750));
+}
+
+uint8_t* RTU::formatFrame(uint8_t* buffer, int address, int functionID,
+                     std::vector<uint8_t> const& payload) {
+
+    return RTU::formatFrame(buffer, address, functionID,
+                            &(*payload.begin()), &(*payload.end()));
```

Comment:
Dereferencing the end iterator is undefined behavior

Reply:
:+1: